_posts/2025-08-06-gpt-oss.md

```diff
@@ -0,0 +1,81 @@
+
+### **Efficient Attention**
+
+GPT-OSS has a highly efficient attention design. It uses GQA with 64 query heads and 8 KV heads. Importantly, the model interleaves full attention and sliding window attention (with window size **128**) with 1:1 ratio. Furthermore, the head size of the model is 64, 50% of the standard head size 128. Finally, each query head has a trained “attention sink” vector.
+
+To efficiently support this attention, vLLM has integrated special GPU kernels from FlashInfer (Blackwell) and FlashAttention 3 (Hopper). Also, we enhanced our Triton attention kernel to support this on AMD GPUs.
+
+Furthermore, to efficiently manage the KV cache with different types of attention (i.e., full and sliding window), vLLM has integrated the [hybrid KV cache allocator](https://arxiv.org/abs/2503.18292), a novel technique proposed by the vLLM team. With the hybrid KV cache manager, vLLM can dynamically share the KV cache space between the full attention layers and sliding window attention layers, reducing the potential memory fragmentation down to zero. 
+
+### **Built-in Tool Support: Agent Loop & Tool Server via MCP**
+
+GPT-OSS includes built-in support for powerful tools, such as web browsing and Python code interpreter. When enabled, the model autonomously decides when and how to invoke these tools, interpreting the results seamlessly.
+
+vLLM natively supports these capabilities by integrating the [OpenAI Responses API](https://platform.openai.com/docs/api-reference/responses) and the gpt-oss toolkit. Through this integration, vLLM implements a loop to parse the model’s tool call, actually invoke the search and code interpreter tools, parse their outputs, and send them back to the model.
+
+Alternatively, users can launch an MCP-compliant external tool server, to let vLLM use the tool server instead of directly leveraging the gpt-oss toolkit. This modular architecture simplifies the creation of scalable tool-calling libraries and services, requiring no internal changes to vLLM.
+
+### **Looking Ahead**
+
+This announcement is just the beginning of vLLM’s continued optimization for GPT-OSS. Our ongoing roadmap includes:
+
+* Hardening the Responses API
+
+* Further enhancing attention DP and MoE EP support
+
+* Reducing CPU overhead to maximize throughput
+
+## Acknowledgement
+
+vLLM team members who contributed to this effort are: Yongye Zhu, Woosuk Kwon, Chen Zhang, Simon Mo, Kaichao You.
+
+Jay Shah from Colfax International implemented the necessary changes to adapt to attention sinks and uncovered optimizations in the FA3 algorithm for gpt-oss. 
+
+We want to thank OpenAI for the amazing partnership: Zhuohan Li, Xiaoxuan Liu, Philippe Tillet, Mario Lezcano-Casado, Dominik Kundel, Casey Dvorak. 
```

Comment:
Could you add Vol Kyrylov?

Reply:
added in [79ba945](https://github.com/vllm-project/vllm-project.github.io/pull/63/commits/79ba945c3ba46ad7460c1683460e92ceb07e8940)